src/haz3lcore/statics/Statics.re

```diff
@@ -636,3 +636,63 @@ let collect_errors = (map: Map.t): list((Id.t, Info.error)) =>
     map,
     [],
   );
+
+/*
+ * Unwrap the input funtion's args (more than 1) into format check and types
+ */
+let caf_argtrans = (args: list(UPat.t)): option(list(UTyp.t)) =>
+  List.fold_right(
+    (item: UPat.t, partial: option(list(UTyp.t))) =>
+      switch (partial) {
+      | Some(lst_typ) =>
+        switch (item.term) {
+        | TypeAnn({ids: _, term: Var(_)}, t) => Some([t, ...lst_typ])
+        | Var(_) => Some([{ids: item.ids, term: EmptyHole}, ...lst_typ])
+        | _ => None
+        }
+      | _ => None
+      },
+    args,
+    Some([]),
+  );
+
+let check_annotated_function_helper =
+    (pat: UPat.t, ret_type: UTyp.t): option((Var.t, UPat.t, UTyp.t)) =>
+  switch (pat.term) {
+  | Ap({ids: _, term: Var(func_name)}, args) =>
+    let in_type_opt: option(UTyp.t) =
```

Comment:
we don't need to check the syntax of the argument -- any argument pattern is fine.